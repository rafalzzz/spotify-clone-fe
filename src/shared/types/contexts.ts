import { NotificationInstance } from 'antd/es/notification/interface';

export type TNotificationContext = {
  api: NotificationInstance;
};
