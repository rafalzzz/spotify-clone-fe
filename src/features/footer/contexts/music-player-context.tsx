'use client';
import React, {
  createContext,
  useState,
  useMemo,
  useContext,
  useRef,
  PropsWithChildren,
  FC,
} from 'react';

import { TMusicPlayerContext } from '../types';

const MusicPlayerContext = createContext<TMusicPlayerContext | undefined>(undefined);

export const MusicPlayerContextProvider: FC<PropsWithChildren> = ({ children }) => {
  const [currentTime, setCurrentTime] = useState(0);
  const [isShuffle, setIsShuffle] = useState(false);
  const [isLoop, setIsLoop] = useState(false);

  const ref = useRef<HTMLAudioElement>(null);

  const memoizedValue = useMemo(
    () => ({
      ref,
      currentTime,
      isShuffle,
      isLoop,
      setIsShuffle,
      setIsLoop,
      setCurrentTime,
    }),
    [currentTime, isLoop, isShuffle],
  );

  return (
    <MusicPlayerContext.Provider value={memoizedValue}>{children}</MusicPlayerContext.Provider>
  );
};

export const useMusicPlayerContext = (): TMusicPlayerContext => {
  const ctx = useContext(MusicPlayerContext);
  if (ctx === undefined) {
    throw new Error('useMusicPlayerContext must be used within a MusicPlayerContextProvider');
  }

  return ctx;
};
