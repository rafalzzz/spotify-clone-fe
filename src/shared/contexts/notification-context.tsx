'use client';
import React, { createContext, useMemo, useContext, FC, PropsWithChildren } from 'react';

import { TNotificationContext } from '@/types/contexts';

const NofiticationContext = createContext<TNotificationContext | undefined>(undefined);

export const NotificationContextProvider: FC<PropsWithChildren<TNotificationContext>> = ({
  children,
  api,
}) => {
  const memoizedValue = useMemo(
    () => ({
      api,
    }),
    [api],
  );

  return (
    <NofiticationContext.Provider value={memoizedValue}>{children}</NofiticationContext.Provider>
  );
};

export const useNotificationContext = (): TNotificationContext => {
  const ctx = useContext(NofiticationContext);
  if (ctx === undefined) {
    throw new Error('useNotificationContext must be used within a NotificationContextProvider');
  }

  return ctx;
};
