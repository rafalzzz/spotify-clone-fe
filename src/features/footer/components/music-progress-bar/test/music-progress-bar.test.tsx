import { render } from '@testing-library/react';
import React from 'react';

import { MusicPlayerContextProvider } from '@/footer/contexts/music-player-context';

import { MusicProgressBar } from '../';

describe('MusicProgressBar', () => {
  it('renders without errors', () => {
    const screen = render(
      <MusicPlayerContextProvider>
        <MusicProgressBar />
      </MusicPlayerContextProvider>,
    );

    expect(screen).toMatchSnapshot();
  });
});
