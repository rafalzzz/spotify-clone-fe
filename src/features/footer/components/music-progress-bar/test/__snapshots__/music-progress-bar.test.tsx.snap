// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicProgressBar renders without errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="music-progress-bar"
      >
        <audio />
        <button
          class="duration duration--text-align-end"
          data-testid="duration-button"
          disabled=""
        >
          0:00
        </button>
        <div
          class="progress-bar"
        >
          <input
            class="progress-bar__input"
            max="0"
            min="0"
            step="0.01"
            type="range"
            value="0"
          />
          <span
            class="progress-bar__span-thumb"
            style="left: -3px;"
          />
        </div>
        <button
          class="duration duration--text-align-start"
          data-testid="duration-button"
        >
          0:00
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="music-progress-bar"
    >
      <audio />
      <button
        class="duration duration--text-align-end"
        data-testid="duration-button"
        disabled=""
      >
        0:00
      </button>
      <div
        class="progress-bar"
      >
        <input
          class="progress-bar__input"
          max="0"
          min="0"
          step="0.01"
          type="range"
          value="0"
        />
        <span
          class="progress-bar__span-thumb"
          style="left: -3px;"
        />
      </div>
      <button
        class="duration duration--text-align-start"
        data-testid="duration-button"
      >
        0:00
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
